docs/reference/argocd.md

```diff
@@ -957,10 +957,133 @@ spec:
+
+!!! warning 
+    `resourceCustomizations` is being deprecated so is encouraged to use `resourceHealthChecks`, `resourceIgnoreDifferences`, and `resourceActions` instead. It is the user's responsibility to not provide conflicting resources if they choose to use both methods of resource customizations. 
+
+### Resource Customizations (with subkeys) Example
+
+Keys for `resourceHealthChecks`, `resourceIgnoreDifferences`, and `resourceActions` are in the form (respectively): `resource.customizations.health.<group_kind>`, `resource.customizations.ignoreDifferences.<group_kind>`, and `resource.customizations.actions.<group_kind>`. The following example defines a custom health check, custom action, and an ignoreDifferences config in the `argocd-cm` ConfigMap. 
+
+``` yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: resource-customizations
+spec:
+  resourceHealthChecks:
+    - group: certmanager.k8s.io
+      kind: Certificate
+      check: |
+        hs = {}
+        if obj.status ~= nil then
+          if obj.status.conditions ~= nil then
+            for i, condition in ipairs(obj.status.conditions) do
+              if condition.type == "Ready" and condition.status == "False" then
+                hs.status = "Degraded"
+                hs.message = condition.message
+                return hs
+              end
+              if condition.type == "Ready" and condition.status == "True" then
+                hs.status = "Healthy"
+                hs.message = condition.message
+                return hs
+              end
+            end
+          end
+        end
+        hs.status = "Progressing"
+        hs.message = "Waiting for certificate"
+        return hs
+  resourceActions:
+  - group: apps
+    kind: Deployment
+    action: |
+      discovery.lua: |
+        actions = {}
+        actions["restart"] = {}
+        return actions
+      definitions:
+        - name: restart
+          # Lua Script to modify the obj
+          action.lua: |
+            local os = require("os")
+            if obj.spec.template.metadata == nil then
+                obj.spec.template.metadata = {}
+            end
+            if obj.spec.template.metadata.annotations == nil then
+                obj.spec.template.metadata.annotations = {}
+            end
+            obj.spec.template.metadata.annotations["kubectl.kubernetes.io/restartedAt"] = os.date("!%Y-%m-%dT%XZ")
+            return obj
+  resourceIgnoreDifferences:
+  - group: admissionregistration.k8s.io
+    kind: MutatingWebhookConfiguration
+    jqPathExpressions: |
+      jsonPointers: |
```

Comment:
It does look like there are other ignore differences customization we can do, namely `managedFieldsManagers` and `jsonPointers` See https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/#system-level-configuration.  Also, It is possible to configure ignoreDifferences to be applied to all resources by setting this key `resource.customizations.ignoreDifferences.all`.  Is it supported/tested by the operator?

Reply:
> It does look like there are other ignore differences customization we can do, namely managedFieldsManagers and jsonPointers See https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/#system-level-configuration

I think those are all behind a `|` so users should be able to specify them if they want and we would just take them as is and paste them here 
That was the reason the 3rd fields were all originally just called `Customization` instead of `check` and `action` and `jqexpressionpath`, maybe we should go back to that 

 >resource.customizations.ignoreDifferences.all. Is it supported/tested by the operator?
 
 not at present to my knowledge, this would need additional changes but should be straight forward